File: Models/Cpu.cs
using System.Text.Json.Serialization;

namespace SpaceInvadersUI.Models
{
    public struct Cpu
    {

        [JsonInclude]
        public byte Opcode;

        [JsonInclude]
        public string Id;

        [JsonInclude]
        public CpuState State;
    }
}

File: src/models/Cpu.ts
import CpuState from "./CpuState";

export default class Cpu {
    opcode: number;
    state: CpuState;
    id: string;

    constructor(id: string, state: CpuState, opcode: number) {
        this.id = id;
        this.state = state;
        this.opcode = opcode;
    }
}
